### PR TITLE
kinder: add test workflows for 1.21

### PR DESCRIPTION
**Pull request by RA489**
xref https://github.com/kubernetes/kubeadm/issues/2362
test infra PR: https://github.com/kubernetes/test-infra/pull/21643

### CODE REVIEW

File: kinder/ci/workflows/skew-1.21-on-1.20.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-  test grid > https://testgrid.k8s.io/sig-cluster-lifecycle-kubeadm#kubeadm-kinder-latest-on-1.20
+  This workflow tests the proper functioning of kubeadm version v1.21 with Kubernetes v1.20
+  test grid > https://testgrid.k8s.io/sig-cluster-lifecycle-kubeadm#kubeadm-kinder-1-21-on-1-20
   config    > https://git.k8s.io/test-infra/config/jobs/kubernetes/sig-cluster-lifecycle/kubeadm-kinder-x-on-y.yaml
   config    > https://git.k8s.io/test-infra/testgrid/config.yaml
 vars:
-  kubeadmVersion: "{{ resolve `ci/latest` }}"
-  kubeletVersion: "{{ resolve `ci/latest-1.20` }}"
+  kubeadmVersion: "{{ resolve `ci/latest-1.21` }}"
+  kubeletVersion: "{{ resolve `ci/latest-1.21` }}"
```
Comment (neolit123) [inline kinder/ci/workflows/skew-1.21-on-1.20.yaml:9]:
```suggestion
  kubeletVersion: "{{ resolve `ci/latest-1.20` }}"
```

**Review by neolit123 — COMMENTED** [review]
we are missing a couple of new workflows:
- external-ca-1.21.yaml
- patches-1.21.yaml

**we also have the kubelet skew tests which are a bit trickier to update.**

the kubeadm and k8s version there matches, while the skew is for the kubelet version.

we need to rename and update the contents of:
- skew-kubelet-1.19-on-latest.yaml -> skew-kubelet-1.19-on-1.21.yaml
- skew-kubelet-1.20-on-latest.yaml -> skew-kubelet-1.20-on-1.21.yaml

add these new workflows too:
- skew-kubelet-1.20-on-latest.yaml
- skew-kubelet-1.21-on-latest.yaml

**Comment by neolit123** [comment]
due to the increased complexity i plan on writing an tool that automates the file updates.
this is something that SIG Arch recommended to us, too.

File: kinder/ci/workflows/skew-1.21-on-1.20.yaml (hunk 1 above)
Comment (RA489) [reply kinder/ci/workflows/skew-1.21-on-1.20.yaml]: updated.

**Comment by RA489** [comment]
added the missing workflows in separate PR to reduce complexy over here.

**Comment by RA489** [comment]
looking for LGTM

**Comment by neolit123** [comment]
> added the missing workflows in separate PR to reduce complexy over here.

two PRs is confusing and difficult to review. a single PR allows overview of the whole change of old / new jobs.

**Comment by RA489** [comment]
> please combine them:
> * 1 for k/kubeadm
> * 1 for k/test-infra

Combined them to one.

**Review by neolit123 — COMMENTED** [review]
/lgtm
/approve

**Comment by palnabarun** [comment]
@neolit123 @RA489 -- renaming a few workflow files in the repository has resulted in failures of https://testgrid.k8s.io/sig-release-master-informing#kubeadm-kinder-upgrade-1-20-latest&width=90. Can you please fix the workflows?

**Comment by palnabarun** [comment]
Or, the test-infra configuration for the above job needs to be updated.

**Pull request merged**